sensor/FlowCompressionSensor.js

```diff
@@ -308,6 +358,51 @@ class FlowCompressionSensor extends Sensor {
     }
   }
 
+  async buildWanBlockCompressedFlows() {
+    while (this.wanBlockBuilding) {
+      await delay(30 * 1000)
+    }
+    this.wanBlockBuilding = true;
+    log.info(`Going to compress wan block flows`)
+    let completed = false
+    let allFlows = []
+    const now = Date.now() / 1000
+    const options = {
+      ts: now,
+      audit: true,
+      count: 2000,
+      macs: sysManager.getLogicInterfaces().map(i => `${Constants.NS_INTERFACE}:${i.uuid}`)
+    }
+    await rclient.delAsync(this.wanCompressedFlowsKey);
+    while (!completed) {
+      try {
+        const flows = await flowTool.prepareRecentFlows({}, JSON.parse(JSON.stringify(options))) || []
+        if (flows.length < options.count) {
+          completed = true
+        } else {
+          options.ts = flows[flows.length - 1].ts
+        }
+        allFlows = allFlows.concat(flows)
+        if (allFlows.length >= this.maxCount) {
+          // compress and dump flows to redis if it exceed count 
+          // big array of allFlows might cause oom
+          const ts = allFlows[allFlows.length - 1].ts;
+          await this.appendAndSave(ts, await this.compress(allFlows), 'wanBlock')
```

Comment:
All data is kept in array and then sent to zlib, creates a big memory consumption. Using pipe should be a better approach. Same problem in build().